packages/mdl-drawer/README.md

```diff
@@ -0,0 +1,284 @@
+| `mdl-temporary-drawer__header`         | Optional. Add to container node to create a 16:9 drawer header.            |
+| `mdl-temporary-drawer__header-content` | Optional. Add to content node inside `mdl-temporary-drawer__header`.       |
+
+### Using the JS Component
+
+MDL Temporary Drawer ships with a Component / Foundation combo which allows for frameworks to richly integrate the
+correct drawer behaviors into idiomatic components.
+
+#### Including in code
+
+##### ES2015
+
+```javascript
+import {MDLTemporaryDrawer, MDLTemporaryDrawerFoundation} from 'mdl-drawer';
+```
+
+##### CommonJS
+
+```javascript
+const mdlDrawer = require('mdl-drawer');
+const MDLTemporaryDrawer = mdlDrawer.MDLTemporaryDrawer;
+const MDLTemporaryDrawerFoundation = mdlDrawer.MDLTemporaryDrawerFoundation;
+```
+
+##### AMD
+
+```javascript
+require(['path/to/mdl-drawer'], mdlDrawer => {
+  const MDLTemporaryDrawer = mdlDrawer.MDLTemporaryDrawer;
+  const MDLTemporaryDrawerFoundation = mdlDrawer.MDLTemporaryDrawerFoundation;
+});
+```
+
+##### Global
+
+```javascript
+const MDLTemporaryDrawer = mdl.Drawer.MDLTemporaryDrawer;
+const MDLTemporaryDrawerFoundation = mdl.Drawer.MDLTemporaryDrawerFoundation;
+```
+
+#### Fully-automatic: DOM Rendering + Initialization
+
+```javascript
+const root = MDLTemporaryDrawer.buildDom();
+const drawer = MDLTemporaryDrawer.attachTo(root);
+// append root to element, etc...
+```
+
+You can use `MDLTemporaryDrawer.buildDom` to dynamically construct the temporary DOM for you.
+
+#### Using an existing element.
+
+If you do not care about retaining the component instance for the temporary drawer, simply call `attachTo()`
+and pass it a DOM element.  
+
+```javascript
+mdl.TemporaryDrawer.attachTo(document.querySelector('.mdl-temporary-drawer'));
+```
+
+#### Manual Instantiation
+
+Temporary drawers can easily be initialized using their default constructors as well, similar to `attachTo`.
+
+```javascript
+import MDLTemporaryDrawer from 'mdl-temporary-drawer';
+
+const drawer = new MDLTemporaryDrawer(document.querySelector('.mdl-temporary-drawer'));
+```
+
+### Using the Foundation Class
+
+MDL Temporary Drawer ships with an `MDLTemporaryDrawerFoundation` class that external frameworks and libraries can
+use to integrate the component. As with all foundation classes, an adapter object must be provided.
+The adapter for temporary drawers must provide the following functions, with correct signatures:
+
+| Method Signature | Description |
+| --- | --- |
+| `addClass(className: string) => void` | Adds a class to the root element. |
+| `removeClass(className: string) => void` | Removes a class from the root element. |
+| `hasClass(className: string) => boolean` | Returns boolean indicating whether element has a given class. |
+| `hasNecessaryDOM() => boolean` | Returns boolean indicating whether the necessary DOM is present (namely, the drawer container). |
```

Comment:
This may need more of an explanation; as a user reading the docs, it's a bit unclear to me what the "drawer container" is. Do you mean the `mdl-{persistent,temporary}-drawer` element itself?